payments-core/src/main/java/com/stripe/android/paymentsheet/model/SupportedSavedPaymentMethod.kt

```diff
@@ -0,0 +1,71 @@
+package com.stripe.android.paymentsheet.model
+
+import android.content.res.Resources
+import androidx.annotation.DrawableRes
+import com.stripe.android.R
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * Enum defining all types of saved payment method supported on Payment Sheet.
+ *
+ * These are fetched from the
+ * [PaymentMethods API endpoint](https://stripe.com/docs/api/payment_methods/list) for returning
+ * customers.
+ */
+internal enum class SupportedSavedPaymentMethod(
+    val type: PaymentMethod.Type
+) {
+    Card(PaymentMethod.Type.Card),
+    SepaDebit(PaymentMethod.Type.SepaDebit);
+
+    companion object {
+        internal fun fromCode(code: String) =
+            values().firstOrNull { it.type.code == code }
+
+        internal fun isSupported(paymentMethod: PaymentMethod) =
+            when (paymentMethod.type) {
+                PaymentMethod.Type.Card -> paymentMethod.card != null
+                PaymentMethod.Type.SepaDebit -> paymentMethod.sepaDebit != null
+                else -> false
+            }
+    }
+}
+
+@DrawableRes
+internal fun PaymentMethod.getIcon(): Int? = when (type) {
```

Comment:
This is a bit of a global reachable function, it might be worth clarifying that it is getting the payment sheet icon in the function name, maybe getSavedPaymentMethodIcon()

Reply:
same for CardBrand.getIcon()